src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentFormNameLoweringPass.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Razor.Language.Components;
+
+internal sealed class ComponentFormNameLoweringPass : ComponentIntermediateNodePassBase, IRazorOptimizationPass
+{
+    // Run after component lowering pass
+    public override int Order => 50;
+
+    protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentIntermediateNode documentNode)
+    {
+        if (!IsComponentDocument(documentNode))
```

Comment:
Well, `.razor` components have many features that are unavailable in non-components and we don't do that for them either, I think everyone knows there are vast differences. But I would do that if you think that would improve UX - but wouldn't it be more confusing since one feature would report a warning and others wouldn't?

Reply:
Yeah I agree with @jjonescz here. It's consistent with how everything else works so we should keep it the same

Reply:
I can agree with that reasoning, but we may want to consider changing the entire policy here 🙂.